src/autopas/utils/SimilarityFunctions.h

```diff
@@ -28,86 +46,68 @@ template <class Container>
-    cellsPerDimension[i] = ceil(domainSizePerDimension[i] / cellLength);
-  }
-  // Actual number of cells we end up with
-  numberOfCells = cellsPerDimension[0] * cellsPerDimension[1] * cellsPerDimension[2];
+  const auto numberOfParticles = container.getNumberOfParticles();
+  const auto domainDimensions = endCorner - startCorner;
+  const auto domainVolume = domainDimensions[0] * domainDimensions[1] * domainDimensions[2];
+
+  // We scale the dimensions of the domain to bins with volumes which give approximately 10 particles per bin.
+  const auto targetNumberOfBins = std::ceil(numberOfParticles / 10.);
+  const auto targetNumberOfBinsPerDim = std::cbrt((double)targetNumberOfBins);
+  // This is probably not an integer, so we floor to get more than 10 particles per bin than too small bins
+  const auto numberOfBinsPerDim = static_cast<int>(std::floor(targetNumberOfBinsPerDim));
+  const auto numberOfBins = numberOfBinsPerDim * numberOfBinsPerDim * numberOfBinsPerDim;
+  const auto binVolume = domainVolume / (double)numberOfBins;
+  const auto binDimensions = domainDimensions / (double)numberOfBinsPerDim;
+
+  // Calculate the actual number of bins per dimension. The rounding is needed in case the division slightly
+  // underestimates the division. The choice of dimension 0 is arbitrary.
+  const int binsPerDimension = static_cast<int>(std::round(domainDimensions[0] / binDimensions[0]));
 
-  std::vector<size_t> particlesPerCell(numberOfCells, 0);
-  std::vector<double> allVolumes(numberOfCells, 0);
+  std::vector<size_t> particlesPerBin(numberOfBins, 0);
 
-  // add particles accordingly to their cell to get the amount of particles in each cell
+  // add particles accordingly to their bin to get the amount of particles in each bin
   for (auto particleItr = container.begin(autopas::IteratorBehavior::owned); particleItr.isValid(); ++particleItr) {
     const auto &particleLocation = particleItr->getR();
-    std::array<size_t, 3> index = {};
-    for (size_t i = 0; i < particleLocation.size(); i++) {
-      index[i] = particleLocation[i] / cellLength;
-    }
-    const size_t cellIndex = autopas::utils::ThreeDimensionalMapping::threeToOneD(index, cellsPerDimension);
-    particlesPerCell[cellIndex] += 1;
-    // calculate the size of the current cell
-    allVolumes[cellIndex] = 1;
-    for (size_t i = 0; i < cellsPerDimension.size(); ++i) {
-      // the last cell layer has a special size
-      if (index[i] == cellsPerDimension[i] - 1) {
-        allVolumes[cellIndex] *= outerCellSizePerDimension[i];
-      } else {
-        allVolumes[cellIndex] *= cellLength;
-      }
-    }
+
+    const auto binIndex3d = autopas::utils::ArrayMath::floorToInt((particleLocation - startCorner) / binDimensions);
+    const auto binIndex1d = autopas::utils::ThreeDimensionalMapping::threeToOneD(
+        binIndex3d, {binsPerDimension, binsPerDimension, binsPerDimension});
+
+    particlesPerBin[binIndex1d] += 1;
```

Comment:
Should probably use a region iterator here. Wouldn't this lead to errors when the selected region is not the whole domain.